### DISCUSSION TITLE
How do I get the complete source code?

### DISCUSSION THREAD

**Discussion by GitS7** [ImageMagick/ImageMagick#5180] [Development]
In the instructions (https://imagemagick.org/script/install-source.php) for compiling the Windows version I find a source code, but it seems to be incomplete. I am trying to build a development environment using Visual Studio 2022 for Windows.
The package ImageMagick-Windows.zip does not contain all files (example: Magick++.h is missing).
The package resulting from the recommended program CloneRepositories.cmd is also incomplete. 
Is there a safe step by step guide to compiling Imagemagick for Windows?   

![Bildschirmfoto 2022-05-30 um 15 22 31](https://user-images.githubusercontent.com/105055946/171021126-9f347f34-ea96-41d4-b37e-c0e4f7b8d609.png)

![Bildschirmfoto 2022-05-30 um 15 27 10](https://user-images.githubusercontent.com/105055946/171021164-cec02e7d-eefb-4094-9afe-203a20210d76.png)

![Bildschirmfoto 2022-05-30 um 15 45 12](https://user-images.githubusercontent.com/105055946/171021219-3c4c35ed-d3b5-478b-adc7-017634ba8f20.png)

![Bildschirmfoto 2022-05-30 um 15 45 58](https://user-images.githubusercontent.com/105055946/171021247-80a786ff-7bfa-48e6-a5e7-0fc5ca8a12ea.png)

![Bildschirmfoto 2022-05-30 um 15 46 25](https://user-images.githubusercontent.com/105055946/171021310-0a65b7ef-0cc2-448b-b8c5-7612174152de.png)

![Bildschirmfoto 2022-05-30 um 15 48 07](https://user-images.githubusercontent.com/105055946/171021423-6f706260-cf22-4dae-91b5-8a0365733bdc.png)

![Bildschirmfoto 2022-05-30 um 16 00 45](https://user-images.githubusercontent.com/105055946/171021460-c210a3ad-1bcd-4324-8bdd-53780ec9f998.png)

![Bildschirmfoto 2022-05-30 um 16 03 23](https://user-images.githubusercontent.com/105055946/171021494-0edae9c0-00d3-4894-baf0-a42258f9cea6.png)

Next.

![Bildschirmfoto 2022-05-30 um 16 13 10](https://user-images.githubusercontent.com/105055946/171021683-8ac51207-c3e7-4930-a7dd-628d1a74ede0.png)

![Bildschirmfoto 2022-05-30 um 16 22 15](https://user-images.githubusercontent.com/105055946/171021722-9e8fe174-e36a-489a-ba25-f7b061b95cd2.png)

![Bildschirmfoto 2022-05-30 um 16 34 08](https://user-images.githubusercontent.com/105055946/171021757-ef0a1acc-289b-43da-8ab4-edf60281894e.png)

![Bildschirmfoto 2022-05-30 um 16 45 26](https://user-images.githubusercontent.com/105055946/171021812-9c6b807c-7484-40b7-955a-76a822cfd353.png)

![Bildschirmfoto 2022-05-30 um 16 48 25](https://user-images.githubusercontent.com/105055946/171021831-2a2b5b6d-063a-4e18-9dbe-339b02201c81.png)

**Comment by dlemstra** [comment]
That C5105 warning is happening because you are using an old version of the Windows 10 SDK. I would advise you to switch to the most recent version. And it looks like we made a mistake in the creation of the source zip file. This will be resolved in the next release.

**Comment by GitS7** [comment]
It was a fresh installation. See screenshot.
![Bildschirmfoto 2022-06-01 um 10 39 34](https://user-images.githubusercontent.com/105055946/171364094-659d419a-f5f1-444f-903d-bdba76b4a228.png)

The following settings are still required in the https://imagemagick.org/script/advanced-windows-installation.php section. However, these are already active:

![Bildschirmfoto 2022-06-01 um 10 43 49](https://user-images.githubusercontent.com/105055946/171364889-9e3ce615-6dc0-43a8-8d9b-45fba969708e.png)

Can you send me a screenshot of your Visual Studio 2022 configuration?

**Comment by GitS7** [comment]
A request! Since the software exists as binaries and you have successfully compiled them, please let me know how you did it.  Are you using an older development environment? With which configuration?

**Comment by GitS7** [comment]
Update to Visual Studio 2022 17.2.3. Git synvhronized. New minimal configuration for ImageMagic. No result.

![Bildschirmfoto 2022-06-01 um 11 20 58](https://user-images.githubusercontent.com/105055946/171387673-5b4419c7-d73f-4ea3-acb8-2b61cf0a26f2.png)

![Bildschirmfoto 2022-06-01 um 11 24 57](https://user-images.githubusercontent.com/105055946/171387706-f1914643-e1f1-40eb-bfb4-e72fb0512370.png)

![Bildschirmfoto 2022-06-01 um 11 35 52](https://user-images.githubusercontent.com/105055946/171387729-d8c7a2b8-b012-418b-8e78-b582a702606f.png)

![Bildschirmfoto 2022-06-01 um 11 37 15](https://user-images.githubusercontent.com/105055946/171387764-fe32ccb6-92ef-4756-aa33-465b6d0d0fe3.png)

![Bildschirmfoto 2022-06-01 um 11 37 29](https://user-images.githubusercontent.com/105055946/171387804-9a8a398c-09dd-4e46-b17c-b8131190c90d.png)

![Bildschirmfoto 2022-06-01 um 11 37 45](https://user-images.githubusercontent.com/105055946/171387833-ab774811-c750-4593-8006-d31d49b38d9b.png)

![Bildschirmfoto 2022-06-01 um 11 38 01](https://user-images.githubusercontent.com/105055946/171387867-c321075b-87ae-4d3b-aee4-a7a40f4994e6.png)

**Comment by dlemstra** [comment]
Please read my earlier response. It tells you what you need to do.

**Reply by GitS7** [reply]
If you mean "That C5105 warning is happening because you are using an old version of the Windows 10 SDK. I would advise you to switch to the most recent version. And it looks like we made a mistake in the creation of the source zip file. This will be resolved in the next release." obove, I check the Windows 10 SDK version, OK. And I have not used the ZIP version, but a Github synchronization. Or did you mean another instruction?

**Comment by GitS7** [comment]
Success with Windows 10 SDK (**10.0.20348.0**).

![Bildschirmfoto 2022-06-01 um 13 38 29](https://user-images.githubusercontent.com/105055946/171398821-bebb79a9-b8cd-4d70-8151-a431bfd87bb0.png)

![Bildschirmfoto 2022-06-01 um 13 54 02](https://user-images.githubusercontent.com/105055946/171398873-29816461-dfd2-45d9-a576-585ca58e2f21.png)
